python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -82,7 +136,13 @@ def get_udf_return_type(func, df):
     else:
         numba_output_type = output_type
 
-    return numba_output_type
+    scalar_return_type = (
+        numba_output_type
+        if not isinstance(numba_output_type, MaskedType)
+        else numba_output_type.value_type
+    )
+
+    return scalar_return_type
```

Comment:
If you don't like returning directly, at minimum the variable name should be changed to `masked_or_scalar` as per our discussion where the function gets called.
```suggestion
    return (
        numba_output_type
        if not isinstance(numba_output_type, MaskedType)
        else numba_output_type.value_type
    )
```

Reply:
Changed it to just return the object instead.